frontend/src/lib/data.ts

```diff
@@ -37,6 +37,46 @@ export async function fetchEvents(limit: number, idCursor?: number, category?: s
   }
 }
 
+export async function fetchEventsInWeek(weekDate: Date, category?: string) {
+  noStore()
+
+  try {
+    const startOfWeek = new Date(weekDate)
+    console.log(startOfWeek)
+    startOfWeek.setDate(weekDate.getDate() - weekDate.getDay())
```

Comment:
a lot of the date stuff can be done via data-fcn library. maybe consider using it to reduce code